browser/app/js/browser/MainActions.js

```diff
@@ -16,83 +16,151 @@
-            <a href="#" className="feba-btn feba-upload">
+  closeMakeBucket() {
+    if (this.state.makeBucketActive) {
+      this.setState({
+        makeBucketActive: false
+      })
+    } else {
+      this.setState({
+        addNewActive: false
+      })
+    }
+    this.makeBucket.clearBucketName()
+  }
+
+  closeMakeBucketOnEscape(e) {
+    if (this.state.addNewActive) {
+      if (e.keyCode === 27) {
+        this.closeMakeBucket()
+      }
+    }
+  }
+
+  componentDidMount() {
+    document.addEventListener(
+      "keydown",
+      this.closeMakeBucketOnEscape.bind(this),
+      false
+    )
+  }
+
+  render() {
+    const { prefixWritable, uploadFile, checkedObjectsCount } = this.props
+    const loggedIn = web.LoggedIn()
+
+    if (loggedIn || prefixWritable) {
+      return (
+        <React.Fragment>
+          <div
+            className={classNames({
+              "add-new": true,
+              "add-new--active": this.state.addNewActive,
+              "add-new--bucket": this.state.makeBucketActive,
+              "add-new--hidden": checkedObjectsCount > 0
+            })}
+          >
+            <i
+              className="add-new__toggle"
+              onClick={this.toggleAddNew.bind(this)}
+            />
+
+            <label
+              htmlFor="add-new-upload"
+              className="add-new__item add-new__item--upload"
+              data-tip="Upload Objects"
+              data-for="tooltip-main-actions"
+            >
               <input
                 type="file"
-                onChange={onFileUpload}
-                style={{ display: "none" }}
-                id="file-input"
+                onChange={this.onFileUpload.bind(this, uploadFile)}
+                id="add-new-upload"
                 multiple={true}
               />
-              <label htmlFor="file-input">
-                {" "}
-                <i className="fa fa-cloud-upload" />{" "}
-              </label>
-            </a>
-          </OverlayTrigger>
-          {loggedIn && (
-            <OverlayTrigger placement="left" overlay={makeBucketTooltip}>
-              <a
-                href="#"
+            </label>
+            {loggedIn && (
+              <div
                 id="show-make-bucket"
-                className="feba-btn feba-bucket"
-                onClick={e => {
-                  e.preventDefault()
-                  showMakeBucketModal()
-                }}
+                className="add-new__item add-new__item--bucket"
+                data-tip="Create New Bucket"
+                data-for="tooltip-main-actions"
+                onClick={
+                  this.state.makeBucketActive
+                    ? undefined
+                    : this.openMakeBucket.bind(this)
+                }
               >
-                <i className="fa fa-hdd-o" />
-              </a>
-            </OverlayTrigger>
+                <MakeBucketModal onRef={ref => (this.makeBucket = ref)} />
```

Comment:
Instead of using refs and calling a function in a child component to clear the content. we can render the component only when `this.state.makeBucketActive` is true.

`{this.state.makeBucketActive && (<MakeBucketModal />)}`

So when the component re-renders, the content will be cleared automatically, no need for `clearBucketName`